colordeducto-port.patch

```diff
@@ -0,0 +1,71 @@
+diff --git a/ColorDeductoActivity.py b/ColorDeductoActivity.py
+index 3a24aa2..452cf7f 100644
+--- a/ColorDeductoActivity.py
++++ b/ColorDeductoActivity.py
+@@ -20,7 +20,6 @@ from sugar3 import profile
+ from sugar3.graphics.toolbarbox import ToolbarBox
+ from sugar3.activity.widgets import ActivityToolbarButton
+ from sugar3.activity.widgets import StopButton
+-from sugar3.graphics.objectchooser import ObjectChooser
+ 
+ from toolbar_utils import button_factory, label_factory, separator_factory
+ from utils import json_load, json_dump
+@@ -32,6 +31,9 @@ from dbus.gi_service import ExportedGObject
+ from sugar3.presence import presenceservice
+ from sugar3.presence.tubeconn import TubeConnection
+ 
++from sugarapp.widgets import SugarCompatibleActivity
++from sugarapp.widgets import DesktopOpenChooser
++
+ from gettext import gettext as _
+ 
+ from game import Game, LEVELS_TRUE, LEVELS_FALSE
+@@ -45,7 +47,7 @@ IFACE = SERVICE
+ PATH = '/in/seeta/ColorDeductoActivity'
+ 
+ 
+-class ColorDeductoActivity(activity.Activity):
++class ColorDeductoActivity(SugarCompatibleActivity):
+     """ Logic puzzle game """
+ 
+ 
+@@ -280,32 +282,19 @@ class ColorDeductoActivity(activity.Activity):
+         ''' Read the Python code from the Journal object '''
+         self._python_code = None
+         try:
+-            _logger.debug("opening %s " % dsobject.file_path)
+-            file_handle = open(dsobject.file_path, "r")
++            _logger.debug("opening %s " % dsobject)
++            file_handle = open(dsobject, "r")
+             self._python_code = file_handle.read()
+             file_handle.close()
+         except IOError:
+-            _logger.debug("couldn't open %s" % dsobject.file_path)
++            _logger.debug("couldn't open %s" % dsobject)
+ 
+     def _chooser(self, filter, action):
+         ''' Choose an object from the datastore and take some action '''
+-        chooser = None
+-        try:
+-            chooser = ObjectChooser(parent=self, what_filter=filter)
+-        except TypeError:
+-            chooser = ObjectChooser(
+-                None, self,
+-                Gtk.DialogFlags.MODAL | Gtk.DialogFlags.DESTROY_WITH_PARENT)
+-        if chooser is not None:
+-            try:
+-                result = chooser.run()
+-                if result == Gtk.ResponseType.ACCEPT:
+-                    dsobject = chooser.get_selected_object()
+-                    action(dsobject)
+-                    dsobject.destroy()
+-            finally:
+-                chooser.destroy()
+-                del chooser
++        chooser = DesktopOpenChooser(self)
++        filename = chooser.get_filename()
```

Comment:
oh, one more thing! I think this is supposed to be a Python file, add a filter for it ?

Reply:
Ok, I didn'y have any clue on how to use it, will add the filter